### DISCUSSION TITLE
Command-Line Windows Install : error NETSDK1135: SupportedOSPlatformVersion cannot be higher than TargetPlatformVersion

### DISCUSSION THREAD

**Discussion by geronimus** [dotnet/maui#6858] [Q&A] [answered]
The [macOS command-line install instructions](https://github.com/dotnet/maui/wiki/macOS-Install) work perfectly.

But the same thing does not work on Windows:
```pwsh
> winget install Microsoft.dotnet
  ...
> dotnet --version
6.0.202
> dotnet workload install maui --source https://api.nuget.org/v3/index.json
  ...
> dotnet new maui -n "MyMauiApp"
The template ".NET MAUI App (Preview)" was created successfully.
> cd MyMauiApp
> dotnet build -t:Run -f net6.0-windows
C:\Program Files\dotnet\sdk\6.0.202\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(897,5): error MSB3073: The
command "path\to\app\bin\Debug\net6.0-windows\win10-x64\MyMauiApp.exe " exited with code 3.

Build FAILED.

> dotnet build -f net6.0-windows
C:\Program Files\dotnet\sdk\6.0.202\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(188,5)
: error NETSDK1135: SupportedOSPlatformVersion 10.0.17763.0 cannot be higher than TargetPlatformVersion 7.

Build FAILED.
```

What am I doing wrong?

Or perhaps a better question is: How can I give `dotnet build` the info it's apparently expecting to get from VisualStudio without actually installing VisualStudio?

**Comment by rogerbriggen** [comment]
You need dotnet 6.0.300, 6.0.202 is not supported by latest maui

**Reply by geronimus** [reply]
I used `winget` to install the latest preview version of dotnet. (Which was `7.0.100-preview.3.22179.4`. There was no option for `6.0.300`.) I got:
```
To build this project, the following workloads must be installed: wasm-tools
```

Okay. Patience. `dotnet workload install wasm-tools`

And now, when I try to build:
```
NETSKD1139: The target platform identifier android was not recognized.
```

So I put `net6.0-windows10.0.19041` in my `.csproj` file's `TargetFrameworks` element and removed everything else in the colon-separated list. I got:
```
MSB4086: A numeric comparison was attempted on "$(registry:HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion@CurrentVersion)" that evaluates to "" instead of a number, in condition "'$(AppxUseResourceIndexerApi)' == '' AND '$(registry:HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion@CurrentVersion)' < '6.3'".
```

At this point, I collapsed to the floor and curled into the fetal position.

> So, why don't you just give yourself a break and use Visual Studio?

So glad you asked!

Perhaps...
- I've tried, but my company Windows device won't let me turn on Developer Mode. Or maybe...
- The licensing terms don't really work for my situation. But really...
- Visual Studio hides the build process, meaning that when problems like the above arise, I'm completely mystified about what's causing them. Or conversely, when things do work, I don't understand why they worked. It's all just "magic".

**Reply by rogerbriggen** [reply]
dotnet 7 does currently not support MAUI...

**Reply by rogerbriggen** [reply]
check https://github.com/dotnet/installer#installers-and-binaries

**Reply by geronimus** [reply]
This is great information! Thank you.

And I think I'm getting there. But...

I installed dotnet version 6.0.400-preview.22257.2.

I then installed the .NET MAUI workloads with `dotnet workload install maui`.

I then created the template project with `dotnet new maui -n "MyMauiApp"`.

And in the new project's folder, I created the recommended Nuget.Config file.

When I attempt `dotnet build`, I now get:
```
error XA5300: The Android SDK directory could not be found. Check that the Android SDK Manager in Visual Studio shows a valid installation. To use a custom SDK path for a command line build, set the 'AndroidSdkDirectory' MSBuild property to the custom path.
```

Since I have no intention of deploying to Android, do I need the Android SDK? Is there a setting I can use to cancel the Android build? Or do I have to download the Android SDK and set the property in my `.csproj` file?

**Reply by geronimus** [reply]
Update:
I think I found an Android SDK already installed at: `C:\Program Files (x86)\Android\android-sdk`

However, when I try any of the following, I still get the above error:
- `dotnet build  -property:AndroidSdkDirectory="C:\Program Files (x86)\Android\android-sdk"`
- `dotnet msbuild  -property:AndroidSdkDirectory="C:\Program Files (x86)\Android\android-sdk"`
- Adding an `AndroidSdkDirectory` element to the property group in my `.csproj` file with the path value.
- Setting `$env:AndroidSdkDirectory="C:\Program Files (x86)\Android\android-sdk"` and running the build

**Comment by rogerbriggen** [comment]
I don't know how to provide the path but maybe you can just remove android from the targetFrameworks in your .csproj file.
Or try
dotnet build -f net6.0-windows10.0.19041.0

**Comment by geronimus** [comment, marked as answer]
It appears that the problem is this issue:
https://github.com/dotnet/maui/issues/5975

Hoping very much that this will be resolved before general availability! (Or shortly thereafter.)
